components/src/dynamo/common/tests/test_video_utils.py

```
@@ -154,3 +162,276 @@ def test_v2_api_fallback_writes_all_frames(self):
+        num_frames, height, width = 30, 144, 176  # QCIF
+        frames = _synthetic_video(num_frames, height, width)
+        try:
+            data = encode_video(frames, fps=30, container="mp4", hw_accel="cpu")
+        except RuntimeError as e:
+            pytest.skip(f"CPU H.264 encoder unavailable: {e}")
+
+        assert b"ftyp" in data[:64]  # mp4 container magic
+
+        try:
+            decoded = iio.imread(data, index=None, extension=".mp4")
+        except Exception as e:  # decoder/plugin not available in this env
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)